### DISCUSSION TITLE
which version of tensorflow is used in federatescope?

### DISCUSSION THREAD

**Discussion by kang-jsnu** [alibaba/FederatedScope#126] [Q&A] [answered]
when i run python federatedscope/main.py --cfg federatedscope/example_configs/femnist.yaml 

report bug:
AttributeError: module 'tensorflow' has no attribute 'set_random_seed'

so i want to know the tensorflow version. i can not find in requirments.txt

**Comment by kang-jsnu** [comment]
In my environment, tensorflow version is 2.9.1.  it is ok.
but you must install pytorch rightly.
i install pytorch GPU version again.

**Comment by xieyxclack** [comment, marked as answer]
The provided example in federatedscope/cross_backend is based on tf.1.12.0 :)
